src/app/pages/component-sidenav/component-sidenav.html

```diff
@@ -4,7 +4,7 @@
               *ngIf="(isScreenSmall | async)"
               [opened]="!(isScreenSmall | async)"
               [mode]="(isScreenSmall | async) ? 'over' : 'side'"
-              [fixedInViewport]="(isScreenSmall | async)"
+              [fixedInViewport]="(isScreenSmall | async) ? true : false"
```

Comment:
This is required with Ivy enabled due to
```ts
component-sidenav.html(7,15): Type 'boolean | null' is not assignable to type 'boolean | undefined'.
```
In `MatSidenav` it appears to have the type of `'boolean | undefined'`
```ts
@Input()
get fixedInViewport(): boolean { return this._fixedInViewport; }
set fixedInViewport(value) { this._fixedInViewport = coerceBooleanProperty(value); }
private _fixedInViewport = false;
```
From `boolean-property.ts`:
```ts
export function coerceBooleanProperty(value: any): boolean {
  return value != null && `${value}` !== 'false';
}
```
In `AsyncPipe`, the type is `'boolean | null'`
```ts
transform<T>(obj: Observable<T> | null | undefined): T | null;
```
@jelbourn do you think that our boolean `@Input()`s and `coerceBooleanProperty()` need to be updated to `'boolean | null'` instead of `'boolean'` (which is implicitly `'boolean | undefined'`)?

Reply:
For now this is fine; I've previously proposed a set of features to the framework team that would solve this more ergonomically, but until then we'll just deal with the annoying behavior